Application/EdFi.Ods.Api/Helpers/AssemblyLoaderHelper.cs

```diff
@@ -29,30 +27,40 @@ namespace EdFi.Ods.Api.Helpers
     public static class AssemblyLoaderHelper
     {
         private static readonly ILog _logger = LogManager.GetLogger(typeof(AssemblyLoaderHelper));
-        private static readonly Dictionary<bool, bool> _assembliesAlreadyLoadedByIncludeFrameworkOption = new();
+        private static readonly Dictionary<Tuple<string, bool>, bool> _assembliesAlreadyLoadedByIncludeFrameworkOption = new();
 
         private const string AssemblyMetadataSearchString = "assemblyMetadata.json";
-        
-        public static void LoadAssembliesFromExecutingFolder(bool includeFramework = false)
+
+        /// <summary>
+        /// Load assemblies from the folder specified by <paramref name="directory"/>.
+        /// If <paramref name="directory"/> is null or empty, the folder containing the currently executing assembly will be used.
+        /// </summary>
+        /// <param name="directory">Optional: Directory from which assemblies should be loaded (Default: null). </param>
+        /// <param name="includeFramework">Optional: Should .NET framework assemblies be included in the loading (Default: false).</param>
+        public static void LoadAssembliesFromFolder(string directory = null, bool includeFramework = false)
         {
-            if (_assembliesAlreadyLoadedByIncludeFrameworkOption.ContainsKey(includeFramework))
+            directory = GetDefaultDirectoryIfNotProvided(directory);
+
+            if (string.IsNullOrEmpty(directory) || !Directory.Exists(directory))
+            {
+                return;
+            }
+
+            var directoryInfo = new DirectoryInfo(directory);
+
+            if (_assembliesAlreadyLoadedByIncludeFrameworkOption.ContainsKey(
+                    new Tuple<string, bool>(directoryInfo.FullName, includeFramework)))

```

Comment:
If you switch to the C# syntax around `ValueTuple` (as described above), you'll just need to drop the `new Tuple<string, bool>` from this code.